node/service/src/block_sealing.rs

```diff
@@ -62,23 +64,26 @@ pub fn frequency_dev_sealing(
 
 	// Start off-chain workers if enabled
 	if config.offchain_worker.enabled {
-		let offchain_workers = Arc::new(sc_offchain::OffchainWorkers::new_with_options(
-			client.clone(),
-			sc_offchain::OffchainWorkerOptions { enable_http_requests: false },
-		));
+		let offchain_workers =
+			sc_offchain::OffchainWorkers::new(sc_offchain::OffchainWorkerOptions {
+				runtime_api_provider: client.clone(),
+				is_validator: config.role.is_authority(),
+				keystore: Some(keystore_container.keystore()),
+				offchain_db: backend.offchain_storage(),
+				transaction_pool: Some(OffchainTransactionPoolFactory::new(
+					transaction_pool.clone(),
+				)),
+				network_provider: network.clone(),
+				enable_http_requests: false,
+				custom_extensions: |_| vec![],
+			});
 
 		// Spawn a task to handle off-chain notifications.
 		// This task is responsible for processing off-chain events or data for the blockchain.
 		task_manager.spawn_handle().spawn(
 			"offchain-notifications",
```

Comment:
nit: It is a bit weird that the offchain-workers runner is categorized under `offchain-notifications` name even though From cumulus I can see that the name for it is `offchain-workers-runner`